### DISCUSSION TITLE
Silence detection and AMD

### DISCUSSION THREAD

**Discussion by Javdu10** [baresip/baresip#3044] [Q&A] [answered]
Hello,

I could not find what I am looking for and chatgpt was giving me weird answers...

I was wondering if there is any documented way to "wait for silence"

Currently I can make a call and play an audio file.

However when it goes to voicemail the audio message is truncated because it does not wait for the recording 

I am reading the output of baresip when I see "Call Established" I play the file.

I am just looking for direction, maybe I could read the bytebuffer of input aubridge ?

The best would be :
- /dial
- `-e "wait_silence of 1second"`
- /ausrc aufile,xxx

OR

- /dial
- read out put and wait for a detected silence
- /ausrc aufile,xxx

Thank you for your time 🙏

**Comment by Javdu10** [comment]
A little update :

I am trying to do a very basic Answering Machine Detection mechanism 

I manage to have something to play with the module vumeter, had to redirect stderr to stdout and regex stdout to get the volume meters based on how it is printed by the module

I could not do a simple readLine because the module only replace characters at a specific position in the last line of stdout

Would have been easier if it had simple prints with new lines

Came up with very simple checks, seems to cover most of the cases

If call established < 2.5 seconds
   Voicemail
If call established > 20 seconds
   Voicemail 

When Voicemail I read every 0.5 seconds the sdtout and wait that it contains "===" for seven loops in a row (waiting for the voicemail message and covers the beep)

Let me know if you have better solutions (using baresip, modules and reading std) 

Cheers

**Comment by cspiel1** [comment, marked as answer]
A much simpler solution would be to
- write your own module based on vumeter
- use `module_event()` to emit your own specific event when a certain condition is met

Your module could go to [baresip-apps](https://github.com/baresip/baresip-apps)

**Reply by Javdu10** [reply]
Indeed, this is the path forward if I cannot make something work with the available tools

So far, I manage to parse stdin and fetch the values I need

Thank you for your reply ✌️
